test/unit/xml/PrettyPrinterSpec.scala

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package unit.xml
+
+import uk.gov.hmrc.customs.api.common.xml.PrettyPrinter
+import util.UnitSpec
+
+class PrettyPrinterSpec extends UnitSpec {
+
+  "PrettyPrinter" should {
+    "format xml with spaces inside elements" in {
+
+      val prettyXml = PrettyPrinter.formatXml(<PreviousDocument><CategoryCode>Z</CategoryCode><ID>ABCDCDS03111026        02</ID><TypeCode>MCR</TypeCode></PreviousDocument>)
+      prettyXml shouldBe
+        """<PreviousDocument>
+          |  <CategoryCode>Z</CategoryCode>
+          |  <ID>ABCDCDS03111026        02</ID>
+          |  <TypeCode>MCR</TypeCode>
+          |</PreviousDocument>
+          |""".stripMargin
+    }
+
+    "format xml with spaces inside attributes" in {
+
+      val prettyXml = PrettyPrinter.formatXml(<PreviousDocument abc="123 ABC"><CategoryCode>Z</CategoryCode><ID>ABCDCDS03111026        02</ID><TypeCode>MCR</TypeCode></PreviousDocument>)
```

Comment:
Better to have multiple spaces in the attributes for this test? Name of test says "spaces" after all.